isisd/isis_zebra.c

```diff
@@ -110,6 +110,9 @@ static int isis_zebra_if_address_add(ZAPI_CALLBACK_ARGS)
 			isis_circuit_add_addr(circuit, c);
 	}
 
+	if (HAS_LINK_PARAMS(c->ifp))
+		isis_mpls_te_update(c->ifp);
+
```

Comment:
Not sure that this is necessary. First, because link parameters are not modified when just modify the address of the interface. Second, because the link parameters are automatically read when the isis circuit is updated and serve to fulfill the LSP. Can you tell me in which case / circumstance it is necessary to call `isis_mpls_te_update()` here?

And, if it is necessary, why not do the same for the `isis_zebra_if_address_del()` function below?

Reply:
I also note that it is the same behavior for ospf (look at `ospf_interface_address_add()` in ospf_zebra.c) which not call `ospf_mpls_te_update()`. So, if it is needed for IS-IS, I think that it would be good if we add it also for OSPF.

Reply:
Actually, in order for the ospf te topotest to pass, this PR updates ospf_interface_address_add to call ospf_mpls_te_update_if.

If it is necessary, then I can update isis_zebra_if_address_del to call ospf_mpls_te_update_if.

I can remove the call to isis_mpls_te_update here, if this call does not make sense.

Reply:
I removed the call to isis_mpls_te_update from isis_zebra_if_address_add.  
(This was a misunderstanding on my part.)

I updated both ospf_ifp_create and ospf_ifp_up to call ospf_mpls_te_update_if.
When just ospf_ifp_create was updated to call ospf_mpls_te_update_if, topotest ospf_te_topo1 failed.

Reply:
Yes. We must be sure that Link Params are ready to be transmitted especially when zebra and ospf start